### PR TITLE
Add source documentation across the board.

### PR DESCRIPTION
**Pull request by k3zi**

### CODE REVIEW

File: Sources/Kumo/Data/MultipartForm.swift [hunk 1]
```diff
@@ -8,33 +8,56 @@ fileprivate func crlf(_ encoding: String.Encoding, count: Int = 1) -> Data {
...
     public let encoding: String.Encoding
+
     let boundary = String(format: "----com.Duet.CNS\(UUID().uuidString)")
-    
+
+    /// The data representation of the multipart form.
     public var data: Data {
         return currentFormData
-            + "--\(boundary)--".data(using: .utf8)!
+            + "--\(boundary)--".data(using: encoding)!
```
Comment (powerje) [inline Sources/Kumo/Data/MultipartForm.swift:23]: 😅

File: Sources/Kumo/HTTP/HTTPRequest.swift [hunk 2]
```diff
@@ -4,20 +4,23 @@ import Foundation
...
 
-public protocol RequestMethod { }
-public protocol RequestResource { }
-public protocol RequestBody { }
-public protocol RequestParameters { }
-public protocol ResponseNestedKey { }
-public typealias RequestOption = RequestMethod & RequestResource & RequestBody & RequestParameters & ResponseNestedKey
-public enum _NoOption: RequestOption { }
-public enum _HasOption: RequestOption { }
+public protocol _RequestMethod { }
```
Comment (powerje) [inline Sources/Kumo/HTTP/HTTPRequest.swift:7]: These changes are breaking, we should make sure to bump the major version

File: Sources/Kumo/Kumo.docc/BlobCache.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+let publisher: AnyPublisher<UIImage, Error> = blobCache.fetch(from: imageURL)
+let dataPublisher: AnyPublisher<Data, Error> = blobCache.fetch(from: imageURL)
+```
+
+A blob cache can also be constructed with just a base URL. A service will
+automatically be created for the blob cache:
+
+```swift
+let blobCache = BlobCache(baseURL: URL(string: "https://www.apple.com")!)
```
Comment (powerje) [inline Sources/Kumo/Kumo.docc/BlobCache.md:41]: 😎

**Review by powerje — APPROVED** [review]

**Review by powerje — CHANGES_REQUESTED** [review]
Please bump the major version then I think we're good to go

**Comment by k3zi** [comment]
@powerje 
> Please bump the major version then I think we're good to go
I believe the version is just controlled by the tag.

**Comment by powerje** [comment]
> 
> 
> I believe the version is just controlled by the tag.
> 
> 

Doh, I thought it was in the podspec as well. Been awhile since I've used Cocoapods 😅

**Review by powerje — APPROVED** [review]

**Comment by k3zi** [comment]
@powerje No, you're right. I just completely forgot about CocaPods.

**Pull request merged**